From cd3b13e0103045310ace4b02d9c4f88e6e14d71c Mon Sep 17 00:00:00 2001
From: "Reddy, Alavala Srinivasa" <alavala.srinivasa.reddy@intel.com>
Date: Fri, 10 Nov 2023 14:49:06 +0530
Subject: [PATCH] Filterout isa perf libraries from vndk list

Currently we have added perf libs under vndksp,
filter them out as they are not new libraries,
rather avx2 implementations of existing libraries.
Should have no impact on GSI.

Signed-off-by: bodapati <shalini.salomi.bodapati@intel.com>
---
 cc/vndk.go | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/cc/vndk.go b/cc/vndk.go
index bb96548e2..3e32e8691 100644
--- a/cc/vndk.go
+++ b/cc/vndk.go
@@ -871,6 +871,15 @@ func getVndkFileName(m *Module) (string, error) {
 	return "", fmt.Errorf("VNDK library should have libraryDecorator or prebuiltLibraryLinker as linker: %T", m.linker)
 }
 
+func filterOutAvxLibs(libList []string) (filtered []string) {
+               for _, lib := range libList {
+                       if !strings.Contains(lib, "_avx2.") {
+                               filtered = append(filtered, lib)
+                       }
+               }
+               return
+       }
+
 func (c *vndkSnapshotSingleton) buildVndkLibrariesTxtFiles(ctx android.SingletonContext) {
 	// Build list of vndk libs as merged & tagged & filter-out(libclang_rt):
 	// Since each target have different set of libclang_rt.* files,
@@ -882,7 +891,9 @@ func (c *vndkSnapshotSingleton) buildVndkLibrariesTxtFiles(ctx android.Singleton
 	_, vndkproduct := vndkModuleListRemover(vndkProductLibraries, "libclang_rt.")(ctx)
 	var merged []string
 	merged = append(merged, addPrefix(llndk, "LLNDK: ")...)
-	merged = append(merged, addPrefix(vndksp, "VNDK-SP: ")...)
+      	// Currently we have added perf libs under vndksp, filter them out as they are not new libraries,
+       	// rather avx implementations of existing libraries. Should have no impact on GSI
+        merged = append(merged, addPrefix(filterOutAvxLibs(vndksp), "VNDK-SP: ")...)
 	merged = append(merged, addPrefix(vndkcore, "VNDK-core: ")...)
 	merged = append(merged, addPrefix(vndkprivate, "VNDK-private: ")...)
 	merged = append(merged, addPrefix(vndkproduct, "VNDK-product: ")...)
-- 
2.17.1

